server/expression/executor_test.go

```diff
@@ -373,6 +373,48 @@ func TestArrayExecution(t *testing.T) {
 			Query:      `[31,35,39] contains 42`,
 			ShouldPass: false,
 		},
+		{
+			Name:       "should_identify_array_instead_of_json",
+			Query:      `["{}", "{}", "{}"] | type = "array"`,
+			ShouldPass: true,
+		},
+	}
+
+	executeTestCases(t, testCases)
+}
+
+func TestJSONExecution(t *testing.T) {
+	testCases := []executorTestCase{
+		{
+			Name:       "should_identify_json_input",
+			Query:      `'{"name": "john", "age": 32, "email": "john@company.com"}' | type = "json"`,
+			ShouldPass: true,
+		},
+		{
+			Name:       "should_be_able_to_compare_with_subset",
+			Query:      `'{"name": "john", "age": 32, "email": "john@company.com"}' contains '{"email": "john@company.com", "name": "john"}'`,
+			ShouldPass: true,
+		},
```

Comment:
I would add another case here as a sanity check: comparing the same object, but with different attribute orders, like: 
```
{"name": "john", "age": 32, "email": "john@company.com"}' contains '{"email": "john@company.com", "name": "john", "age": "32"}
```